config/logit/20_custom_cf_filters.conf

```diff
@@ -110,3 +110,19 @@ mutate {
     "[gorouter][bytesreceived]" => "integer"
   }
 }
+
+# Process x_forwarded_for in gorouter logs in same way as done for RTR logs above
+if [gorouter][header][x_forwarded_for] {
+  # Set [gorouter][header][x_forwarded_for]
+  mutate {
+    gsub => ["[gorouter][header][x_forwarded_for]","[\s\"]",""] # remove quotes and whitespace
+    split => ["[gorouter][header][x_forwarded_for]", ","] # format is client, proxy1, proxy2 ...
+  }
+
+  # Set [gorouter][remote_addr]
+  # NOTE this value could potentially be spoofed - correct x-forwarded-for
+  # handling in logstash is very tricky
+  mutate {
+    add_field => ["[gorouter][remote_addr]", "%{[gorouter][header][x_forwarded_for][0]}"]
```

Comment:
Could we make sure the field added has [the IP type](https://www.elastic.co/guide/en/elasticsearch/reference/current/ip.html)? That way we'll be able to do all the [types of query](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-iprange-aggregation.html) that are supported on that type  

https://discuss.elastic.co/t/how-to-assign-a-new-field-a-certain-type-and-value/34148/2

Reply:
Did not know about that at all

Reply:
Looks like `ip` is not a valid value https://www.elastic.co/guide/en/logstash/current/plugins-filters-mutate.html#plugins-filters-mutate-convert, even if it exists as a type in ES.

Reply:
Bugger. Oh well, let's get this shipped with the known limitations. Better than nothing.